test/testHelpers.hpp

```diff
@@ -34,6 +35,16 @@
 #include <mmio.h>
 #endif
 
+bool operator==(const af_half &lhs, const af_half &rhs) {
+    return lhs.data_ == rhs.data_;
+}
+
+std::ostream &operator<<(std::ostream &os, const af_half &val) {
```

Comment:
Do we use `half_float::half` from `extern/half/include/half.hpp` only in tests ?

Reply:
Yes. Our half type is not fully functional. It would require us to reimplement the entire half library inorder to correctly implement it. I don't think we should directly include other libraries in our API. perhaps we can discuss it in a separate issue.